packages/darklang/stdlib/string.dark

```diff
@@ -157,15 +159,24 @@ module Darklang =
 
       /// Checks if <param lookingIn> contains <param searchingFor>
       let contains (lookingIn: String) (searchingFor: String) : Bool =
-        match Builtin.String.indexOf lookingIn searchingFor with
-        | Some _ -> true
-        | None -> false
+        PACKAGE.Darklang.Stdlib.Option.isSome (Builtin.String.indexOf lookingIn searchingFor)
 
 
       /// Returns the substring of <param string> between the <param from> and <param to> indices.
       /// Negative indices start counting from the end of <param string>.
       let slice (string: String) (from: Int) (``to``: Int) : String =
-        Builtin.String.slice_v0 string from ``to``
+        let len = Builtin.String.length string
+        let normalize =  (fun i ->
+          (if i < 0 then len + i else i)
+          |> PACKAGE.Darklang.Stdlib.Int.min len
+          |> PACKAGE.Darklang.Stdlib.Int.max 0)
+
+        let f = normalize from
+        let l = normalize ``to``
+        let l = if f > l then f else l
+
+        (PACKAGE.Darklang.Stdlib.String.dropLast string (len - l))
```

Comment:
TODO: find a different way to implement slice, dropLast uses slice (infinite recursion)